cadc-gms/src/main/java/org/opencadc/auth/StandardIdentityManager.java

```diff
@@ -0,0 +1,272 @@
+        }
+    }
+
+    @Override
+    public Subject validate(Subject subject) throws NotAuthenticatedException {
+        validateOidcAccessToken(subject);
+        return subject;
+    }
+
+    @Override
+    public Subject augment(Subject subject) {
+        // TODO: if X500Principal && servops && UMS we could augment CADC-style
+        // oidc tokens: validate gets HttpPrincipal and NumericPrincial
+        // cadc signed cookies/tokens: validate gets all identities
+        return subject;
+    }
+
+    @Override
+    public Subject toSubject(Object owner) {
+        Subject ret = new Subject();
+        if (owner != null) {
+            UUID uuid = null;
+            if (owner instanceof UUID) {
+                uuid = (UUID) owner;
+            } else if (owner instanceof String) {
+                String sub = (String) owner;
+                uuid = UUID.fromString(sub);
+            } else {
+                throw new RuntimeException("unexpected owner type: " + owner.getClass().getName() + " value: " + owner);
+            }
+            NumericPrincipal p = new NumericPrincipal(uuid);
+            
+            // effectively augment by using the current subject as a "cache" of known identities
+            Subject s = AuthenticationUtil.getCurrentSubject();
+            if (s != null) {
+                for (Principal cp : s.getPrincipals()) {
+                    if (AuthenticationUtil.equals(p, cp)) {
+                        log.debug("[cache hit] caller Subject matches " + p + ": " + s);
+                        ret.getPrincipals().addAll(s.getPrincipals());
+                        return ret;
+                    }
+                }
+            }
+            
+            ret.getPrincipals().add(p);
+            // TODO: this is sufficient for some purposes, but not for output using toDisplayString
+            // a real augment would require system credentials to implement
+        }
+        return ret;
+    }
+
+    @Override
+    public Object toOwner(Subject subject) {
+        // use NumericPrincipal aka OIDC sub for persistence
+        Set<NumericPrincipal> ps = subject.getPrincipals(NumericPrincipal.class);
+        if (ps.isEmpty()) {
+            return null;
+        }
+        return ps.iterator().next().getUUID().toString();
+    }
+
+    @Override
+    public String toDisplayString(Subject subject) {
+        // prefer HttpPrincipal aka OIDC preferred_username for string output, eg logging
+        Set<HttpPrincipal> ps = subject.getPrincipals(HttpPrincipal.class);
+        if (!ps.isEmpty()) {
+            return ps.iterator().next().getName(); // kind of ugh
+        }
+        
+        // try X509
+        Set<X500Principal> px = subject.getPrincipals(X500Principal.class);
+        if (!px.isEmpty()) {
+            return px.iterator().next().getName(); // kind of ugh
+        }
+        
+        return null;
```

Comment:
Don't want to use getName() of whatever else might be in the subject (instead of null)?

Reply:
Aside from HttpPrincipal or X500, which other one can there actually be? Don't think one wants to output NumericPrincipal here... this is for logging,  x-vo-authenticated header output, job ownerID field output, etc.